### ISSUE TITLE
Session Copy Modal: Instructors able to select the Course that they are copying from

### ISSUE THREAD

**Issue by rexong** [TEAMMATES/teammates#12552]
**Environment**: 
- `master` branch at commit `8633d4e`

**Steps to reproduce**
Given that there is a session call `First team feedback session (percentage-based)` from `test.tes-demo` course. 
1. Navigate to Instructor Session Page. 
2. Click on the `Copy` button of `First team feedback session (percentage-based)`
3. Select `test.tes-demo` as the course to copy to.

<ins>Video Demo</ins>

https://github.com/TEAMMATES/teammates/assets/66376253/2fcaaa8e-3f13-4e89-9395-a5538faaa443

**Current behaviour**
Error will be shown stating that the session is already in the course.
![image](https://github.com/TEAMMATES/teammates/assets/66376253/facef50e-e802-4ded-9405-1e3ddb3a0204)

**Suggestions**
Perhaps the instructor should not be allowed to copy the session into the same course. 
This could be resolved by either
1. Remove the selection for the same course, ie not displaying the same course as a choice
2. Disable the selection for the same course, ie displaying the same course as a choice, but not allowing instructors to select it.

**Comment by domlimm** [comment]
@rexong Fair point! This lies in the UX, or maybe the relationship between a Course and Feedback Session. The suggestions that you've provided are good. We could also allow them to select the same session but if the same session already exists, perhaps we could have them make some changes for differentiation.

My understanding might not be 100% accurate on this. Let's hear what the rest thinks 👍🏻

**Comment by rexong** [comment]
I am clearer about the use case now. 
However, I still think the UX here can still be improved. 

Currently, if the user did not change the session name and tried to copy the session again, the error toast pops up stating that the current session already exist in the course. 
However, they are not instructed to change the session name so that they can copy the session to the same course successfully.
In this case, I would suggest that we can consider the following ideas to improve the UX.
1. Disable the checkbox, only if session name is changed would the checkbox be enabled. 
2. When copying to the same course, if the session name is not changed, the error toast can also indicate that the session name can be changed. 

Would like to hear your input! @damithc @domlimm

**Comment by damithc** [comment]
@rexong 

> However, I still think the UX here can still be improved.

I agree.


> * Disable the checkbox, only if session name is changed would the checkbox be enabled.

This could be misleading, as it gives the impression that the action is not allowed.

> * When copying to the same course, if the session name is not changed, the error toast can also indicate that the session name can be changed.

Do you mean when the user clicks the `Copy` button, check if the same course has been chosen but the the session name remains the same, and inform the user that the name needs to change (while still staying in the dialog)? Yes, that could be helpful, as the user can now fix the problem faster and fewer steps, compared to the current behavior.

**Comment by rexong** [comment]
@damithc 
> Do you mean when the user clicks the `Copy` button, check if the same course has been chosen but the the session name remains the same, and inform the user that the name needs to change (while still staying in the dialog)? Yes, that could be helpful, as the user can now fix the problem faster and fewer steps, compared to the current behavior.

I was initially thinking of informing the user using the error toast (not staying in the dialog). 
But I prefer the idea of informing the user while staying in the dialog like you suggested.
